test/array.cpp

```diff
@@ -409,13 +409,24 @@ TEST(array_reshape_3d) {
 }
 
 TEST(array_negative_strides) {
+  // Test with a negative stride in y.
   array_of_rank<int, 2> a({{0, 10, 3}, {0, 3, -1}});
-  ASSERT_EQ(a.x().stride(), 3);
   for_all_indices(a.shape(), [&](int x, int y) { a(x, y) = y; });
```

Comment:
Say something like:

Initialize a(x, y) = y and check that it's still true if you make a compact copy that makes strides positive).

(and again for x).

Reply:
Rewrote this to use fill_pattern/check_pattern. I'm not sure why I didn't do that in the first place. Funny thing about codebases that only get a few commits every few months :)